pkgs/common-updater/combinators.nix

```diff
@@ -120,21 +120,28 @@ rec {
     let
       scripts = scriptsNormalized;
       hasCommitSupport = lib.findSingle ({ supportedFeatures, ... }: supportedFeatures == [ "commit" ]) null null scripts != null;
+      hasSilentSupport = lib.findFirst ({ supportedFeatures, ... }: supportedFeatures == [ "silent" ]) null scripts != null;
       validateFeatures =
         if hasCommitSupport then
           ({ supportedFeatures, ... }: supportedFeatures == [ "commit" ] || supportedFeatures == [ "silent" ])
+        else if hasSilentSupport then
+          ({ supportedFeatures, ... }: supportedFeatures == [ "silent" ])
         else
```

Comment:
```suggestion
        else
          # No update script declares any supported feature to fail loudly on unknown features rather than silently discard them.
```